client/my-sites/email/email-management/list/email-management-list-item.jsx

```diff
@@ -0,0 +1,358 @@
+
+	recordAddEmailEvent( accountType ) {
+		const { domain } = this.props;
+		recordTracksEvent( 'calypso_email_management_add_email_account', {
+			domain: domain.name,
+			account_type: accountType,
+		} );
+	}
+
+	canUserAddEmailAccounts( domain ) {
+		const { user } = this.props;
+
+		if ( hasTitanMailWithUs( domain ) ) {
+			// TODO: Remove this check once the back end support for this is in place.
+			const titanOwnerId = getTitanMailOwnerId( domain );
+			if ( titanOwnerId === null ) {
+				return true;
+			}
+			return titanOwnerId === user.ID;
+		}
+
+		if ( hasGSuiteWithUs( domain ) ) {
+			return getGSuiteOwnerId( domain ) === user.ID;
+		}
+
+		if ( hasEmailForwards( domain ) ) {
+			/* TODO: Update the logic */
+			return this.props.userCanManageOptions;
+		}
+
+		return false;
+	}
+
+	getHeaderImage( domain ) {
+		const { translate } = this.props;
+
+		if ( hasTitanMailWithUs( domain ) ) {
+			return (
+				<Gridicon
+					className="email-management-list-item__header-icon titan"
+					icon="my-sites"
+					size={ 36 }
+				/>
+			);
+		}
+
+		if ( hasGSuiteWithUs( domain ) ) {
+			return (
+				<img
+					className="email-management-list-item__header-icon google"
+					src={ googleWorkspaceIcon }
+					alt={ translate( 'Google Workspace icon' ) }
+				/>
+			);
+		}
+
+		if ( hasEmailForwards( domain ) ) {
+			return (
+				<img
+					className="email-management-list-item__header-icon email-forward"
+					src={ emailForwardingIcon }
+					alt={ translate( 'Email Forwarding icon' ) }
+				/>
+			);
+		}
+
+		return null;
+	}
+
+	getAddEmailHandler( domain ) {
+		if ( hasTitanMailWithUs( domain ) ) {
+			return this.onAddTitanMailbox;
+		}
+
+		if ( hasGSuiteWithUs( domain ) ) {
+			return this.onAddGoogleUser;
+		}
+
+		if ( hasEmailForwards( domain ) ) {
+			return this.onAddEmailForward;
+		}
+
+		return null;
+	}
+
+	getEmailProductName( domain ) {
+		if ( hasTitanMailWithUs( domain ) ) {
+			return getTitanProductName();
+		}
+
+		if ( hasGSuiteWithUs( domain ) ) {
+			return domain.productName ?? getGoogleMailServiceFamily();
+		}
+
+		if ( hasEmailForwards( domain ) ) {
+			return this.props.translate( 'Email Forwarding' );
+		}
+
+		return null;
+	}
+
+	getMailboxCount( domain ) {
+		if ( hasTitanMailWithUs( domain ) ) {
+			return getMaxTitanMailboxCount( domain );
+		}
+
+		if ( hasGSuiteWithUs( domain ) ) {
+			return getGSuiteMailboxCount( domain );
+		}
+
+		if ( hasEmailForwards( domain ) ) {
+			return getEmailForwardsCount( domain );
+		}
+
+		return 0;
+	}
+
+	renderFoldableContent() {
+		const { domain, translate } = this.props;
+
+		const addEmailHandler = this.getAddEmailHandler( domain );
+		const wrappedAddEmailHandler = ( event ) => {
+			event.preventDefault();
+			addEmailHandler();
+		};
+
+		return (
+			<React.Fragment>
+				{ this.renderEmails() }
+				{ this.canUserAddEmailAccounts( domain ) && (
+					<VerticalNavItem path="#" onClick={ wrappedAddEmailHandler }>
+						{ translate( 'Add new mailbox' ) }
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Add New Mailboxes' )` *ES Score: 8*